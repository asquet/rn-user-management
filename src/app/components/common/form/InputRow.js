import React from 'react';
import { View, TextInput, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  wrapper: {
    height: 50,
    padding: 3,
  },
  input: {
    flex: 1,
  },
});

export default function InputRow({ value, label, onChange, wrapperStyles, children, ...props }) {
  return (
    <View style={[styles.wrapper, wrapperStyles]}>
      {
        children ||
        <TextInput
          style={styles.input}
          placeholder={label}
          onChangeText={onChange}
          value={value}
          {...props}
        />
      }
    </View>
  );
}

InputRow.propTypes = {
  value: React.PropTypes.string,
  label: React.PropTypes.string.isRequired,
  onChange: React.PropTypes.func,
  children: React.PropTypes.node,
  wrapperStyles: React.PropTypes.shape({}),
};

InputRow.defaultProps = {
  onChange: () => {},
  value: '',
  children: null,
  wrapperStyles: null,
};
