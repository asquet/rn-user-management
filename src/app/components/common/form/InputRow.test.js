import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import InputRow from './InputRow';

// Note: test renderer must be required after react-native.

describe('InputRow', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <InputRow label="InputRow" />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
