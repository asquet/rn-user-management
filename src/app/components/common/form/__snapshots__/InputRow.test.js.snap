// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRow renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 50,
        "padding": 3,
      },
      null,
    ]
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="InputRow"
    style={
      Object {
        "flex": 1,
      }
    }
    value=""
  />
</View>
`;
